vizro-core/src/vizro/models/_base.py

```diff
@@ -101,8 +108,116 @@ def _is_discriminated_union(field):
         cls.update_forward_refs(**vm.__dict__.copy())
         new_type.update_forward_refs(**vm.__dict__.copy())
 
+    def dict(self, **kwargs):
+        # Overwrite pydantic's own `dict` method to add __vizro_model__ to the dictionary.
+        # To get exclude as an argument is a bit fiddly because this function is called recursively
+        # inside pydantic, which sets exclude=None by default.
+        if kwargs.get("exclude") is None:
+            kwargs["exclude"] = self.__vizro_exclude_fields__()
+        _dict = super().dict(**kwargs)
+        # Inject __vizro_model__ into the dictionary - needed just for to_python.
+        _dict["__vizro_model__"] = self.__class__.__name__
+        return _dict
+
+    # This is like pydantic's own __exclude_fields__ but safer to use (it looks like __exclude_fields__ no longer
+    # exists in pydantic v2).
+    # Root validators with pre=True are always included, even when exclude_default=True, and so this is needed
+    # to potentially exclude fields set this way, like Page.id.
+    def __vizro_exclude_fields__(self) -> Optional[Union[Set[str], Mapping[str, Any]]]:
+        return None
+
+    def _to_python(self, extra_imports: Optional[Set[str]] = None, extra_callable_defs: Set[str] = set()) -> str:
+        """Converts a Vizro model to the Python code that would create it.
+
+        Args:
+            extra_imports: Extra imports to add to the Python code. Provide as a set of complete import strings.
+            extra_callable_defs: Extra callable definitions to add to the Python code. Provide as a set of complete
+                function definitions.
+
+        Returns:
+            str: Python code to create the Vizro model.
+
+        """
+        # Model
+        model_dict = self.dict(exclude_unset=True)
+        model_code = "model = " + _dict_to_python(model_dict)
+
+        # Imports
+        extra_imports_concat = "\n".join(extra_imports) if extra_imports else None
+
+        # CapturedCallable definitions
+        callable_defs_set = _extract_captured_callable_source() | extra_callable_defs
+        callable_defs_concat = "\n".join(callable_defs_set) if callable_defs_set else None
+
+        # Data Manager
+        data_defs_list = _extract_captured_callable_data_info()
+        data_defs_concat = "\n".join(data_defs_list) if data_defs_list else None
```

Comment:
Minor suggestion: To enhance readability and maintain consistency, consider reordering the sections of the code as follows:

- #Imports
- #CapturedCallable Definitions
- #Data Manager
- #Model

If you agree, please adjust the order of the `_concatenate_code` arguments and the `TO_PYTHON_TEMPLATE.format` arguments accordingly to reflect this structure.
```py
TO_PYTHON_TEMPLATE.format(
        extra_imports=extra_imports if extra_imports else "",
        callable_defs=callable_defs,
        data_settings=data_settings,
        code=code,
)
```

Reply:
Good point, implemented the new order